bufferoverride/gui/bufferoverrideview.cpp

```diff
@@ -192,6 +202,61 @@ void BufferOverrideView::draw(VSTGUI::CDrawContext *ctx) {
     }
   }
 
+  // draw the time-scale legend
+  constexpr CCoord LEGEND_MARGIN_TOP = 7;
+  constexpr CCoord LEGEND_MARGIN_BOTTOM = 8;
+  const auto legend_height = MARGIN_BOTTOM -
+    (LEGEND_MARGIN_TOP + LEGEND_MARGIN_BOTTOM);
+  assert(legend_height > 0);
+  const auto legend_top = height - LEGEND_MARGIN_BOTTOM - legend_height;
+  const auto legend_bottom = legend_top + legend_height;
+  constexpr auto legend_left = MARGIN_HORIZ;
+  const auto legend_right = width - MARGIN_HORIZ;
+  const auto legend_width = legend_right - legend_left;
+  constexpr CCoord LEGEND_DASH_STRIDE = 2;
+  constexpr CCoord LEGEND_DASH_WIDTH = 1;
+  const auto legend_dash_y = std::floor(std::midpoint(legend_top,
+                                                      legend_bottom));
+
+  // legend bounding
+  offc->setFrameColor(color_legend);
+  offc->drawLine({legend_left, legend_top},
+                 {legend_left, legend_bottom});
+  offc->drawLine({legend_right, legend_top},
+                 {legend_right, legend_bottom});
+  for (auto x = legend_left + LEGEND_DASH_STRIDE;
+       x <= (legend_right - LEGEND_DASH_STRIDE);
+       x += LEGEND_DASH_STRIDE) {
+    DrawFilledBox(x, legend_dash_y,
+                  LEGEND_DASH_WIDTH, LEGEND_DASH_WIDTH,
+                  color_legend);
+  }
+
+  // legend label
+  const auto legend_label = [](auto window_sec) {
+    if (window_sec < 1) {
+      return std::to_string(std::lround(window_sec * 1000)) + " ms";
+    }
+    return std::to_string(std::lround(window_sec)) + " sec";
+  }(window_sec);
+  constexpr bool LABEL_ANTIALIAS = false;
+  constexpr CCoord LABEL_PADDING = 12;
+  const auto label_width = offc->getStringWidth(legend_label.c_str()) +
+    (LABEL_PADDING * 2);
+  const auto label_x = legend_left +
+    ((legend_width - label_width) / 2);
+  const DGRect text_area(label_x, legend_top + 1, label_width,
+                         std::min<CCoord>(dfx::kFontContainerHeight_Pasement9px,
```

Comment:
I think we should always use the constant for the text container (so just dfx::kFontContainerHeight_Pasement9px) because the workaround tweaks assume this. I think that nothing really goes wrong if legend_height is not tall enough (but also it doesn't matter because it's a constant anyway)?